### PR TITLE
Add service variant support

### LINKED ISSUE #2613: Include frontpage fixtures for WS variants
**Is your feature request related to a problem? Please describe.**
WorldService has a few variant sites, these are different scripts for the same service which have their own distinct URL. We need to include the fixture data for these WS variants for front pages

**Describe the solution you'd like**
Add front page fixture data for:

```
ukchina/simp
ukchina/trad
zhongwen/simp
zhongwen/trad
serbian/lat
serbian/cyr
```

Follow the folder structure of : `<service>/<variant>/fontpage/index,json`

**Describe alternatives you've considered**

**Testing notes**
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

**Additional context**

### LINKED ISSUE #3323: Single service bundle for service variants
**Is your feature request related to a problem? Please describe.**
As of right n ow services variant supports has gone with the idea of creating new bundle for each service variant i.e.

Serbian will have two variant bundles

`serbian_lat.js` & `serbian_cyr.js`

**Describe the solution you'd like**
Can we have a single service bundle for services with variants e.g. `serbian.js` instead of `serbian_lat.js` & `serbian_cyr.js`

The service config could be in one file and have two named exports for each of the variants.

The serviceContext provider could then take a serviceVariant prop to ensure the correct configuration is loaded.

Benefits of this approach being that switching between variants would not need to have a new service bundle downloaded and would actually reduce the complexity of the current service variant work in simorgh.

**Describe alternatives you've considered**
Sticking with the current approach of have a bundle per service or per variant.

**Testing notes**
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [ ] This feature is expected to need manual testing.

**Additional context**

### PR DESCRIPTION
**Pull request by dr3**
### TESTING NOTES: https://github.com/bbc/simorgh/pull/3509#issuecomment-530309419

WIP implementation description https://gist.github.com/dr3/253646ac0bd288e0d1bc7ed74b7dff59

Resolves #2613
Resolves https://github.com/bbc/simorgh/issues/3323

**Overall change:** Add service variant support

**Code changes:**

- Adds a variant key to all service configs, top level key is now 'default', 'simp', 'trad', 'lat' or 'cyr'
- Updates cypress tests to deal with new config
- Adds zhongwen simp, ukchina simp & serbian lat to cypress config
- Updates variant fixture data, with extra text in headline for easy visualisation of variant
- Adds `variant` and `pathname` to app return
- Moves and improve link logic in metadata to request context provider
- Updates service context provider to handle variants
- Creates variant handler for variant logic
- Updates article and frontpage get initial data logic for variants
- Updates regexes for frontpages and article pages for variants

NB: This work only includes frontpages & articles, Media pages wernt added due to them recently being added and not wanting to clash them their development. The small piece of work to add variant support will come later. https://github.com/bbc/simorgh/issues/3699

NB: This PR doesnt include accurate service config for each variant, that will be added when needed

### CODE REVIEW WARNING.
The large number of files in this PR means github doesn't reveil some diffs. Be careful to not skip over changes, especially those at the bottom after the ALL the services 

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [x] This PR requires manual testing

### CODE REVIEW

**Comment by sareh** [comment]
@dr3 I'm about to start code reviewing this - this major change could do with a Readme update to explain how we're handling service variants. Can you please add this? It will aid understanding of the changes for developers and testers working on Simorgh.

File: README.md [hunk 1]
```diff
@@ -153,6 +159,11 @@ The World Service front pages follow the article format for AMP too, being avail
 - [http://localhost.bbc.com:7080/igbo.amp](http://localhost.bbc.com:7080/igbo.amp)
 - [http://localhost.bbc.com:7080/pidgin.amp](http://localhost.bbc.com:7080/pidgin.amp)
 
+Services with variants can't be accessed using the format above, instead the variant must be provided in the URL.
+
+- [http://localhost.bbc.com:7080/zhongwen/simp](http://localhost.bbc.com:7080/zhongwen/simp)
+- [http://localhost.bbc.com:7080/zhongwensimp.amp](http://localhost.bbc.com:7080/zhongwen/simp.amp).
```
Comment (ghost) [inline README.md]:
```suggestion
- [http://localhost.bbc.com:7080/zhongwen/simp.amp](http://localhost.bbc.com:7080/zhongwen/simp.amp).
```

File: cypress/support/helpers/runTestsForPage.js [hunk 2]
```diff
@@ -45,41 +45,35 @@ const runTestsForPage = ({
...
+        const testArgs = {
+          service,
+          pageType,
+          variant: config[service].variant,
+        };
+
         // Enables overriding of the smoke test values in the config/services.js file
-        testsThatAlwaysRunForAllPages({ service, pageType });
-        testsThatAlwaysRunForAllCanonicalPages({ service, pageType });
+        testsThatAlwaysRunForAllPages(testArgs);
```
Comment (chris-hinds) [inline cypress/support/helpers/runTestsForPage.js:55]: I like having the testArgs object here. Any reason this setup wasn't followed in other similar files too?

File: src/app/containers/PageHandlers/withContexts.jsx [hunk 3]
```diff
@@ -35,14 +45,17 @@ const WithContexts = Component => {
     id: string,
     isAmp: bool.isRequired,
     pageType: string.isRequired,
+    pathname: string.isRequired,
     previousPath: string,
     service: string.isRequired,
+    variant: oneOf(['simp', 'trad', 'lat', 'cyr']),
```
Comment (chris-hinds) [inline src/app/containers/PageHandlers/withContexts.jsx]: Might have to extract this out at a later date when we add more styles of variants.

Comment (chris-hinds) [reply src/app/containers/PageHandlers/withContexts.jsx]: Actually you have this same line in two locations now, should this be extracted out somewhere?

File: src/app/containers/App/App.test.jsx [hunk 4]
```diff
@@ -14,7 +14,7 @@ describe('App', () => {
   let wrapper;
   const initialData = { pageData: 'Some initial data' };
   const error = 'Error!';
-  const match = { params: { service: 'news', amp: false } };
+  const match = { params: { service: 'news', amp: false, variant: '/simp' } };
```
Comment (ghost) [inline src/app/containers/App/App.test.jsx:17]:
```suggestion
  const match = { params: { service: 'news', amp: false, variant: 'simp' } };
```

File: src/app/containers/FrontPage/index.test.jsx [hunk 5]
```diff
@@ -88,7 +89,7 @@ describe('FrontPageContainer', () => {
         });
 
         const { useContext, useReducer, useState } = jest.requireMock('react');
-        useContext.mockReturnValue({ ...igboConfig, lang: 'ig' });
+        useContext.mockReturnValue({ ...igboConfig.default, lang: 'ig' });
```
Comment (ghost) [inline src/app/containers/FrontPage/index.test.jsx]:
```suggestion
        useContext.mockReturnValue(igboConfig.default);
```
Igbo now has `lang`

File: src/app/contexts/ServiceContext/index.test.jsx [hunk 6]
```diff
@@ -41,17 +41,33 @@ describe('ServiceContextProvider', () => {
           return <span>{brandName}</span>;
         };
 
+        const serviceContextProps = {
+          service,
+          // Dont pass variant if its 'default', this better mirrors the
+          // behaviour in the production app, where varient is unset for
```
Comment (chris-hinds) [inline src/app/contexts/ServiceContext/index.test.jsx]:
```suggestion
          // behaviour in the production app, where variant is unset for
```

File: src/app/lib/config/services/serbian.js [hunk 7]
```diff
@@ -20,16 +19,14 @@ const service = {
   audioCaptionOffscreenText: 'Audio caption',
   defaultCaptionOffscreenText: 'Caption, ',
   imageCopyrightOffscreenText: 'Image source, ',
-  locale: `sr-Latn`,
-  datetimeLocale: `sr-latn`,
+  datetimeLocale: `sr-Latn`.toLowerCase(),
```
Comment (chris-hinds) [inline src/app/lib/config/services/serbian.js]: we have recently removed all use cases of `toLowerCase()` in the config files, any reason this is back?

File: src/app/lib/config/services/serbian.js [hunk 8]
```diff
@@ -162,4 +163,19 @@ const service = {
   timezone: 'Europe/London',
 };
 
+const service = {
+  lat: {
+    ...baseServiceConfig,
+    lang: `sr-Latn`,
```
Comment (chris-hinds) [inline src/app/lib/config/services/serbian.js]: should this be lowercase?

File: src/app/containers/App/App.test.jsx (hunk 4 above)
Comment (dr3) [reply src/app/containers/App/App.test.jsx:17]: This is needed :) `/simp` is what we get back from the match :)

**Review by chris-hinds — CHANGES_REQUESTED** [review]

File: cypress/support/helpers/runTestsForPage.js (hunk 2 above)
Comment (dr3) [reply cypress/support/helpers/runTestsForPage.js]: This was recently done by @sadickisaac, dont wanna refactor much in this PR due to size :P

Comment (dr3) [reply cypress/support/helpers/runTestsForPage.js:55]: I dont follow what you mean soz? This itterator is quite specific, could you give an example of somewhere else? I mainly just did stuff were the alternative was too messy :P

File: src/app/containers/App/App.test.jsx (hunk 4 above)
Comment (ghost) [reply src/app/containers/App/App.test.jsx:17]: Confirmed `/simp` is returned from the match. Confused this with the sanitised variant.

File: cypress/support/helpers/runTestsForPage.js (hunk 2 above)
Comment (chris-hinds) [reply cypress/support/helpers/runTestsForPage.js:55]: Ignore this....... Ive looked back over it what i was thinking doesn't really make sense.

File: src/app/contexts/RequestContext/getMetaUrls/index.js [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+
+  return canonicalUrl.replace(ampRegex, '');
+};
+
+const getUkCanonicalUrl = url => {
+  const tldRegex = /(.com|.co.uk)/g;
+  return url.replace(tldRegex, '.co.uk');
+};
+
+const getNonUkCanonicalUrl = url => {
```
Comment (AlistairGempf) [inline src/app/contexts/RequestContext/getMetaUrls/index.js:15]: This doesn't necessarily get the canonical URL, you could equally pass the amp url and get the non-UK version of that (similarly for the UK function above).

```suggestion
const getNonUkUrl = url => {
```

Comment (dr3) [reply src/app/contexts/RequestContext/getMetaUrls/index.js:15]: > you could equally pass the amp url

The way this file takes `origin, pathname`, and that we have `getAmpUrl` above, this is more accurate as 'Canonical' IMO, thoughts?

Comment (AlistairGempf) [reply src/app/contexts/RequestContext/getMetaUrls/index.js:15]: To me the function name `getNonUkCanonicalUrl(url)` suggests it'll convert any URL into the non-UK canonical version, but as you say given the way this file is laid out it's fairly explanatory that it won't

**Review by chris-hinds — APPROVED** [review]

File: src/app/contexts/utils/createLoadableContext/fixtures/fooService.js [hunk 10]
```diff
@@ -1,5 +1,8 @@
 const fooService = {
   foobar: 'valueFromFixtureData',
+  simp: {
+    foobar: 'valueFromSimpVarientFixtureData',
```
Comment (sareh) [inline src/app/contexts/utils/createLoadableContext/fixtures/fooService.js]:
```suggestion
    foobar: 'valueFromSimpVariantFixtureData',
```

File: src/app/contexts/utils/createLoadableContext/index.test.jsx [hunk 11]
```diff
@@ -10,25 +10,46 @@ const Component = () => {
...
-    </LoadableServiceProvider>,
-  );
+    const { getByText } = render(
+      <LoadableServiceProvider configKey="simp">
+        <Component />
+      </LoadableServiceProvider>,
+    );
 
-  const expectedFixtureText = 'valueFromFixtureData';
+    const expectedFixtureText = 'valueFromSimpVarientFixtureData';
```
Comment (sareh) [inline src/app/contexts/utils/createLoadableContext/index.test.jsx]:
```suggestion
    const expectedFixtureText = 'valueFromSimpVariantFixtureData';
```

File: cypress/integration/pages/testsForAllPages.js [hunk 12]
```diff
@@ -172,7 +173,7 @@ export const testsThatFollowSmokeTestConfigforAllPages = ({
                     title = '';
                 }
                 /* Note that if updating these, also do the same for errorPage404/test.js */
```
Comment (sareh) [inline cypress/integration/pages/testsForAllPages.js]:
```suggestion
                /* Note that if updating these, also do the same for errorPage404/tests.js */
```

File: package.json [hunk 13]
```diff
@@ -25,7 +25,7 @@
     "test:dependencies": "npx dependency-check@3.4.1 ./package.json",
     "test:e2e": "./scripts/killApp.sh && npm run build && run-p --race start cypress",
     "test:e2e:interactive": "./scripts/killApp.sh && npm run build && run-p --race start cypress:interactive",
-    "test:lint": "eslint --ext .js,jsx,json ./src ./data ./cypress ./.storybook ./webpack** && swagger-cli validate data/schema.yaml && stylelint 'src/**/*.js' 'src/**/*.jsx'",
+    "test:lint": "eslint --fix --ext .js,jsx,json ./src ./data ./cypress ./.storybook ./webpack** && swagger-cli validate data/schema.yaml && stylelint 'src/**/*.js' 'src/**/*.jsx'",
```
Comment (sareh) [inline package.json]: I don't think you meant to update this in this PR, so I'm suggesting this change. 
```suggestion
    "test:lint": "eslint --ext .js,jsx,json ./src ./data ./cypress ./.storybook ./webpack** && swagger-cli validate data/schema.yaml && stylelint 'src/**/*.js' 'src/**/*.jsx'",
```

File: cypress/support/config/services.js [hunk 14]
```diff
@@ -801,16 +829,31 @@ module.exports = {
   },
   serbian: {
     font: undefined,
+    isWorldService: true,
+    variant: 'lat',
```
Comment (sareh) [inline cypress/support/config/services.js:879]: This setup doesn't allow easy addition of multiple variants to the Cypress config.

We'll want cypress tests for both the `lat` and `cyr` variants for the `serbian` service. Likewise for `trad` as well as `simp` variants for `zhongwen` and `ukchina`. 

I'd suggest changing the config shape to have variant value as a key, since the fonts and pageType paths can differ for different variants. 
```
  serbian: {
    isWorldService: true,
    lat: {
      font: undefined,
      pageTypes: {
        articles: {
          path:
            Cypress.env('APP_ENV') === 'live' || Cypress.env('APP_ENV') === 'test'
              ? undefined
              : '/serbian/articles/c805k05kr73o/lat',
          smoke: true,
        },
        errorPage404: {
          path:
            Cypress.env('APP_ENV') === 'live' || Cypress.env('APP_ENV') === 'test'
              ? undefined
              : '/serbian/articles/cabcdefghijo/lat',
          smoke: true,
        },
        frontPage: { path: '/serbian/lat', smoke: true },
        liveRadio: { path: undefined, smoke: false },
      },
    },
    cyr: {
      font: undefined,
      pageTypes: {
        articles: {
          path:
            Cypress.env('APP_ENV') === 'live' || Cypress.env('APP_ENV') === 'test'
              ? undefined
              : '/serbian/articles/c805k05kr73o/cyr',
          smoke: true,
        },
        errorPage404: {
          path:
            Cypress.env('APP_ENV') === 'live' || Cypress.env('APP_ENV') === 'test'
              ? undefined
              : '/serbian/articles/cabcdefghijo/cyr',
          smoke: true,
        },
        frontPage: { path: '/serbian/cyr', smoke: true },
        liveRadio: { path: undefined, smoke: false },
      },
    }
  },
```
A consequence of this restructure would be to need the `default` key for services that don't have variants. e.g.
```
  persian: {
    isWorldService: true,
    default: {
      font: 'Nassim',
      pageTypes: {
        ...
      },
    },
  },

File: src/app/routes/getInitialData/utils/getRouteProps/index.js [hunk 15]
```diff
@@ -12,11 +13,15 @@ const getRouteProps = (routes, url) => {
 
   const amp = path(['amp'], params);
   const service = path(['service'], params);
+  const rawVariant = path(['variant'], params);
```
Comment (sareh) [inline src/app/routes/getInitialData/utils/getRouteProps/index.js]: Renaming as variantPath would be useful for readability since this gives  `/cyr` `/lat` `/simp` `/trad` etc. Had to log locally for this. Also adding a comment here would help.

File: src/app/lib/utilities/variantHandler/index.js [hunk 16]
```diff
@@ -0,0 +1,21 @@
+const servicesWithVariants = {
+  serbian: ['lat', 'cyr'],
+  ukchina: ['simp', 'trad'],
+  zhongwen: ['simp', 'trad'],
+};
+
+// Remove leading slash from variant
+export const variantSanitzer = variant => variant && variant.replace('/', '');
```
Comment (sareh) [inline src/app/lib/utilities/variantHandler/index.js]: Typo here and in all places this is used:
```suggestion
export const variantSanitizer = variant => variant && variant.replace('/', '');
```

File: src/app/lib/utilities/variantHandler/index.js [hunk 17]
```diff
@@ -0,0 +1,21 @@
...
+
+// If service has variants, use it or default to first variant in array
+// If service doesnt have variants, return 'default'
+export const variantDataKey = (service, rawVariant) => {
```
Comment (sareh) [inline src/app/lib/utilities/variantHandler/index.js]: The only place this function is used is in `ServiceContext/index.jsx` and there, the variant value passed in doesn't have the leading `/`. So I don't believe the sanitizer is needed here. I also suggest renaming:
```suggestion
export const getVariant = ({ service, variant }) => {
```

File: cypress/support/config/services.js (hunk 14 above)
Comment (dr3) [reply cypress/support/config/services.js:879]: Chatted with @benjaminhobbs about this, soz should have put in PR. We decided to just test 1 variant for now, and later either add tests that navigate between the 2 variants, or change the config. I think this PR would become much too big If I refactor cypress tests too. Happy top raise an issue?

File: src/app/containers/Metadata/index.jsx [hunk 18]
```diff
@@ -225,7 +183,7 @@ const MetadataContainer = ({ metadata, promo }) => {
         articleAuthor={articleAuthor}
         articleSection={pathOr(null, ['passport', 'genre'], metadata)}
         brandName={brandName}
-        canonicalLink={canonicalLink}
+        canonicalLink={canonicalNonUkLink}
```
Comment (sareh) [inline src/app/containers/Metadata/index.jsx:186]: I believe this changes the canonical link for `/news/articles/*` pages to always use `*bbc.com`. I don't think that is what we want. 
```suggestion
        canonicalLink={canonicalLink}
```

File: src/app/containers/Metadata/index.jsx [hunk 19]
```diff
@@ -207,7 +165,7 @@ const MetadataContainer = ({ metadata, promo }) => {
     <>
       <LinkedData
         brandName={brandName}
-        canonicalLink={canonicalLink}
+        canonicalLink={canonicalNonUkLink}
```
Comment (sareh) [inline src/app/containers/Metadata/index.jsx:168]: Same here
```suggestion
        canonicalLink={canonicalLink}
```

File: src/app/contexts/RequestContext/index.jsx [hunk 20]
```diff
@@ -54,6 +57,7 @@ RequestContextProvider.propTypes = {
   isAmp: bool.isRequired,
   pageType: oneOf(['article', 'frontPage', 'media', 'error']).isRequired,
   service: string.isRequired,
+  pathname: string.isRequired,
```
Comment (sareh) [inline src/app/contexts/RequestContext/index.jsx:63]: What's the reason for not having `variant` as part of the RequestContext?  I think we'll need it pretty soon. For example, for creating a script switcher component that will need to know the current `variant` value.  We'll also need it for stats tracking of components that differ re the variant. 

I suggest adding `variant` to the RequestContext.

File: src/app/contexts/utils/createLoadableContext/fixtures/fooService.js (hunk 10 above)
Comment (dr3) [reply src/app/contexts/utils/createLoadableContext/fixtures/fooService.js]: Done :)

File: src/app/contexts/utils/createLoadableContext/index.test.jsx (hunk 11 above)
Comment (dr3) [reply src/app/contexts/utils/createLoadableContext/index.test.jsx]: Done :)

File: cypress/integration/pages/testsForAllPages.js (hunk 12 above)
Comment (dr3) [reply cypress/integration/pages/testsForAllPages.js]: Done :)

File: package.json (hunk 13 above)
Comment (dr3) [reply package.json]: Done :)

File: src/app/routes/getInitialData/utils/getRouteProps/index.js (hunk 15 above)
Comment (dr3) [reply src/app/routes/getInitialData/utils/getRouteProps/index.js]: Done :)

File: src/app/lib/utilities/variantHandler/index.js (hunk 16 above)
Comment (dr3) [reply src/app/lib/utilities/variantHandler/index.js]: Done :)

File: src/app/lib/utilities/variantHandler/index.js (hunk 17 above)
Comment (dr3) [reply src/app/lib/utilities/variantHandler/index.js]: Done :)

File: src/app/contexts/RequestContext/getMetaUrls/index.test.js [hunk 21]
```diff
@@ -0,0 +1,50 @@
+import getMetaUrls from '.';
+
+const tests = [
+  {
+    origin: 'https://www.bbc.co.uk',
+    pathname: '/foo/bar',
```
Comment (sareh) [inline src/app/contexts/RequestContext/getMetaUrls/index.test.js]: Can you please replace `/foo/bar` with an article example instead? It makes it more useful for developers wanting to understand the complex canonical logic.

File: src/app/contexts/RequestContext/getMetaUrls/index.test.js [hunk 22]
```diff
@@ -0,0 +1,50 @@
...
+      canonicalUkLink: 'https://www.bbc.co.uk/foo/bar',
+      ampUkLink: 'https://www.bbc.co.uk/foo/bar.amp',
+      canonicalNonUkLink: 'https://www.bbc.com/foo/bar',
+      ampNonUkLink: 'https://www.bbc.com/foo/bar.amp',
+    },
+    assertion: 'should return expected output for .com origin and pathname',
+  },
+  {
+    origin: 'https://www.bbc.com',
+    pathname: '/serbian/lat',
```
Comment (sareh) [inline src/app/contexts/RequestContext/getMetaUrls/index.test.js:32]: Please add test examples for `news` as well, since it is an edge case that should have a different `canonicalLink` value depending on the origin. e.g. when the `origin: 'https://www.bbc.co.uk',` and when `origin: 'https://www.bbc.com',`

File: src/server/index.test.jsx [hunk 23]
```diff
@@ -45,9 +54,10 @@ const mockRouteProps = ({ id, service, isAmp, dataResponse, responseType }) => {
   getRouteProps.mockReturnValue({
     isAmp,
     service,
+    variant,
     route: { getInitialData },
     match: {
-      params: { id, service },
+      params: { id, service, variant: variant ? `/${variant}` : undefined },
```
Comment (sareh) [inline src/server/index.test.jsx]: Please add a comment to explain that you've added the leading `/` here to the `variant` since that's how it is captured by our route regex. This is otherwise quite confusing to read. 
Also to add clarity, you can make a mock value and use it here. 
```const mockVariantParam = variant => variant ? `/${variant}` : undefined;```

File: cypress/support/config/services.js (hunk 14 above)
Comment (sareh) [reply cypress/support/config/services.js:879]: Okay, I see. Yes please do raise a separate issue so we can test multiple variants in our cypress tests.

File: src/app/containers/Metadata/index.jsx (hunk 18 above)
Comment (sareh) [reply src/app/containers/Metadata/index.jsx:186]: Seems I was mistaken. Canonical is pinned to use `www.bbc.com`/ `www.test.bbc.com`

File: src/server/index.test.jsx (hunk 23 above)
Comment (dr3) [reply src/server/index.test.jsx]: Done :)

File: src/app/contexts/RequestContext/getMetaUrls/index.test.js (hunk 22 above)
Comment (dr3) [reply src/app/contexts/RequestContext/getMetaUrls/index.test.js:32]: Done :)

File: src/app/contexts/RequestContext/getMetaUrls/index.test.js (hunk 21 above)
Comment (dr3) [reply src/app/contexts/RequestContext/getMetaUrls/index.test.js]: Done :)

File: src/app/contexts/RequestContext/index.jsx (hunk 20 above)
Comment (dr3) [reply src/app/contexts/RequestContext/index.jsx:63]: Done :)

File: cypress/support/config/services.js (hunk 14 above)
Comment (dr3) [reply cypress/support/config/services.js:879]: https://github.com/bbc/simorgh/issues/3698

File: package.json [hunk 24]
```diff
@@ -38,7 +38,7 @@
       "pre-push": "sh scripts/stopPushOnLatest.sh && sh scripts/upToDateWithLatest.sh && sh scripts/enforceVersions.sh && npm run test:unit -- --changedSince=latest && apache2-license-checker"
     }
   },
-  "repository": {
+"repository": {
```
Comment (sareh) [inline package.json]:
```suggestion
  "repository": {
```

File: src/app/routes/regex/index.test.js [hunk 25]
```diff
@@ -64,38 +68,71 @@ describe('articleDataRegexPath', () => {
...
 });
 
 describe('frontpageRegexPath', () => {
-  const validRoutes = ['/news', '/persian', '/news.amp', '/persian.amp'];
+  const validRoutes = [
+    '/news',
+    '/persian',
+    '/news.amp',
+    '/persian.amp',
+    '/news/simp',
```
Comment (sareh) [inline src/app/routes/regex/index.test.js:92]: `/:service/simp` should not be valid for services that are not `ukchina` or `zhongwen`. We should tighten up our regex. Happy for that to be done in a future PR, instead of keeping this one in progress.

File: src/app/lib/config/services/serbian.js [hunk 26]
```diff
@@ -38,7 +35,6 @@ const service = {
     home: 'Home',
     currentPage: 'Current page',
     skipLinkText: 'Skip to content',
-    relatedContent: 'Related content',
```
Comment (sareh) [inline src/app/lib/config/services/serbian.js]: Since you've pulled in `latest` some fields have changed in the serbian config. Can you please change them to match the `latest` branch? 
`relatedContent` and `mostRead` fields are missing and two footer links were added back in again which should be removed.

File: src/app/lib/config/services/ukchina.js [hunk 27]
```diff
@@ -38,7 +36,6 @@ const service = {
     home: 'Home',
     currentPage: 'Current page',
     skipLinkText: 'Skip to content',
-    relatedContent: 'Related content',
```
Comment (sareh) [inline src/app/lib/config/services/ukchina.js]: Same here - `relatedContent` field is missing.

File: src/app/lib/config/services/zhongwen.js [hunk 28]
```diff
@@ -38,7 +36,6 @@ const service = {
     home: 'Home',
     currentPage: 'Current page',
     skipLinkText: 'Skip to content',
-    relatedContent: 'Related content',
```
Comment (sareh) [inline src/app/lib/config/services/zhongwen.js]: Same here `relatedContent` field is missing

**Review by sareh — COMMENTED** [review]
A few more comments. Mostly config changes that were hidden in the diffs.

File: src/app/routes/regex/index.test.js (hunk 25 above)
Comment (dr3) [reply src/app/routes/regex/index.test.js:92]: IMO we should leave it up to the data failing, but happy to discuss that later :)

**Review by sareh — APPROVED** [review]
Thanks for all the changes. Great work!


yaaay

**Comment by dr3** [comment]
Testing notes:
This change is large and affects almost everything. While the aim of this PR is to add variants, the focus of the testing should be to ensure there is no regression on services without variants.

This includes ensuring consistency with latest on metadata tags, ati & chartbeat params and other service specific, detail orientated logic.

Any not critical issues noted with service variants can be noted and we can raise them as issues to fix forward :)

**Comment by PriyaKR** [comment]
Testing:
Canonical and amp version of the below pages
http://localhost.bbc.com:7080/serbian/lat
http://localhost.bbc.com:7080/serbian/cyr
http://localhost.bbc.com:7080/ukchina/trad
http://localhost.bbc.com:7080/ukchina/simp
http://localhost.bbc.com:7080/zhongwen/simp
http://localhost.bbc.com:7080/zhongwen/trad
http://localhost.bbc.com:7080/persian
http://localhost.bbc.com:7080/igbo

http://localhost.bbc.com:7080/news/articles/c6v11qzyv8po
http://localhost.bbc.com:7080/persian/articles/c4vlle3q337o

Done a regression looks good except few issues noted for services with variant
- cookie banner message is still in english
- favicon in page title is missing

Apart from these ATI,chartbeat and also the cypress tests needs a look into.

**Comment by dr3** [comment]
Also theres
http://localhost.bbc.com:7080/serbian/articles/c805k05kr73o/lat
http://localhost.bbc.com:7080/serbian/articles/c805k05kr73o/cyr
http://localhost.bbc.com:7080/ukchina/articles/c0e8weny66ko/simp
http://localhost.bbc.com:7080/ukchina/articles/c0e8weny66ko/trad
http://localhost.bbc.com:7080/zhongwen/articles/c3xd4x9prgyo/simp
http://localhost.bbc.com:7080/zhongwen/articles/c3xd4x9prgyo/trad 
:)

**Comment by staylos92** [comment]
LGTM 👍 

There are a couple of comments above from priya which I believe @dr3 is going to raise as separate issues?

**Comment by dr3** [comment]
>Done a regression looks good except few issues noted for services with variant
>
>- cookie banner message is still in english
>- favicon in page title is missing
>
>Apart from these ATI,chartbeat and also the cypress tests needs a look into.

Have decided these will be picked up as general release work for the services, no specific issues needed

have made
https://github.com/bbc/simorgh/issues/3699
https://github.com/bbc/simorgh/issues/3698
Which cover more general variant issues/work
